backend/x86_proc.ml

```diff
@@ -15,6 +15,70 @@
 
 open X86_ast
 
+module StringTbl = Hashtbl.Make (struct
+    type t = string
+
+    let equal = String.equal
+
+    let hash = Hashtbl.hash
+  end)
+
+module SectionName = struct
+  type t =
+    { name: string list;
+      name_str: string;
+      flags : string option;
+      args: string list
+    }
+
+  let equal t1 t2 =
+    List.equal (String.equal) t1.name t2.name
+
+  let hash t = Hashtbl.hash t.name
+
+  let compare t1 t2 = List.compare (String.compare) t1.name t2.name
+
+  let make name flags args =
+    { name; name_str = String.concat "," name; flags; args; }
+
+  let from_name name = { name = [name]; name_str = name; flags = None; args = [] }
+
+  let name t = t.name_str
+
+  let flags t =
```

Comment:
Can this move into the internal assembler code?  It's ELF-specific, whereas this file is also used for non-ELF x86 targets.